Windows support

Has anyone successfully got this working on Windows? If so, what steps were necessary?

Trying to build in the Docker Windows build container I get the following

```
PS C:\go\src\github.com\asottile\dockerfile> c:\python\python .\setup.py install
running install
running bdist_egg
running egg_info
creating dockerfile.egg-info
writing dockerfile.egg-info\PKG-INFO
writing dependency_links to dockerfile.egg-info\dependency_links.txt
writing top-level names to dockerfile.egg-info\top_level.txt
writing manifest file 'dockerfile.egg-info\SOURCES.txt'
reading manifest file 'dockerfile.egg-info\SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'dockerfile.egg-info\SOURCES.txt'
installing library code to build\bdist.win-amd64\egg
running install_lib
running build_ext
$ GOPATH='C:\Users\ContainerAdministrator\AppData\Local\Temp\tmprkpwgxfo' go get -d
$ CGO_CFLAGS='-Ic:\python\include -Ic:\python\include' CGO_LDFLAGS='-Lc:\python\libs -lpython36' GOPATH='C:\Users\ContainerAdministrator\AppData\Local\Temp\tmprkpwgxfo' go build -buildmode=c-shared -o 'C:\go\src\github.com\asottile\dockerfile\build\lib.win-amd64-3.6\dockerfile.cp36-win_amd64.pyd'
# _/C_/Users/ContainerAdministrator/AppData/Local/Temp/tmprkpwgxfo/src/github.com/asottile/dockerfile/pylib
In file included from c:\python\include/pytime.h:6:0,
                 from c:\python\include/Python.h:68,
                 from support.c:1:
c:\python\include/object.h:910:17: error: initializer element is not constant
 #define Py_None (&_Py_NoneStruct)
                 ^
support.c:14:34: note: in expansion of macro 'Py_None'
 PyObject* PyDockerfile_Py_None = Py_None;
                                  ^
Traceback (most recent call last):
  File ".\setup.py", line 37, in <module>
    setup_requires=['setuptools-golang>=0.2.0'],
  File "c:\python\lib\site-packages\setuptools\__init__.py", line 143, in setup
    return distutils.core.setup(**attrs)
  File "c:\python\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "c:\python\lib\distutils\dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "c:\python\lib\distutils\dist.py", line 974, in run_command
    cmd_obj.run()
  File "c:\python\lib\site-packages\setuptools\command\install.py", line 67, in run
    self.do_egg_install()
  File "c:\python\lib\site-packages\setuptools\command\install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "c:\python\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "c:\python\lib\distutils\dist.py", line 974, in run_command
    cmd_obj.run()
  File "c:\python\lib\site-packages\setuptools\command\bdist_egg.py", line 172, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "c:\python\lib\site-packages\setuptools\command\bdist_egg.py", line 158, in call_command
    self.run_command(cmdname)
  File "c:\python\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "c:\python\lib\distutils\dist.py", line 974, in run_command
    cmd_obj.run()
  File "c:\python\lib\site-packages\setuptools\command\install_lib.py", line 11, in run
    self.build()
  File "c:\python\lib\distutils\command\install_lib.py", line 107, in build
    self.run_command('build_ext')
  File "c:\python\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "c:\python\lib\distutils\dist.py", line 974, in run_command
    cmd_obj.run()
  File "c:\python\lib\site-packages\setuptools\command\build_ext.py", line 78, in run
    _build_ext.run(self)
  File "c:\python\lib\distutils\command\build_ext.py", line 339, in run
    self.build_extensions()
  File "c:\python\lib\distutils\command\build_ext.py", line 448, in build_extensions
    self._build_extensions_serial()
  File "c:\python\lib\distutils\command\build_ext.py", line 473, in _build_extensions_serial
    self.build_extension(ext)
  File "c:\python\lib\site-packages\setuptools_golang.py", line 131, in build_extension
    _check_call(cmd_build, cwd=pkg_path, env=env)
  File "c:\python\lib\site-packages\setuptools_golang.py", line 80, in _check_call
    subprocess.check_call(cmd, cwd=cwd, env=dict(os.environ, **env))
  File "c:\python\lib\subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '('go', 'build', '-buildmode=c-shared', '-o', 'C:\\go\\src\\github.com\\asottile\\dockerfile\\build\\lib.win-amd64-3.6\\dockerfile.cp36-win_amd64.pyd')' returned non-zero exit status 2.
```

--- a/README.md
+++ b/README.md
@@ -1,4 +1,5 @@
 [![Build Status](https://travis-ci.org/asottile/dockerfile.svg?branch=master)](https://travis-ci.org/asottile/dockerfile)
+[![Build status](https://ci.appveyor.com/api/projects/status/l5kj12ysd49xul1l?svg=true)](https://ci.appveyor.com/project/asottile/dockerfile)
 
 dockerfile
 ==========

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,0 +1,17 @@
+environment:
+    matrix:
+    -   PYTHON: 'C:\Python27'
+    -   PYTHON: 'C:\Python37'
+
+install:
+-   'SET PATH=%PYTHON%;%PYTHON%\Scripts;C:\MinGW\bin;C:\go-x86\bin;%PATH%'
+-   'SET GOROOT=C:\go-x86'
+-   git submodule update --init
+-   pip install pytest .
+
+# Not a C# project
+build: false
+
+test_script: pytest tests
+
+cache: '%LOCALAPPDATA%\pip\cache'

--- a/pylib/main.go
+++ b/pylib/main.go
@@ -4,7 +4,7 @@ package main
 // #include <Python.h>
 //
 // extern int PyDockerfile_PyArg_ParseTuple_U(PyObject*, PyObject**);
-// extern PyObject* PyDockerfile_Py_None;
+// extern PyObject* PyDockerfile_Py_RETURN_NONE();
 //
 // extern PyObject* PyDockerfile_GoIOError;
 // extern PyObject* PyDockerfile_GoParseError;
@@ -46,8 +46,7 @@ func stringToPy(s string) *C.PyObject {
 
 func stringToPyOrNone(s string) *C.PyObject {
 	if s == "" {
-		C.Py_IncRef(C.PyDockerfile_Py_None)
-		return C.PyDockerfile_Py_None
+		return C.PyDockerfile_Py_RETURN_NONE()
 	} else {
 		return stringToPy(s)
 	}

--- a/pylib/support.c
+++ b/pylib/support.c
@@ -11,7 +11,9 @@ int PyDockerfile_PyArg_ParseTuple_U(PyObject* args, PyObject** s) {
 }
 
 /* go cannot access C macros */
-PyObject* PyDockerfile_Py_None = Py_None;
+PyObject* PyDockerfile_Py_RETURN_NONE() {
+    Py_RETURN_NONE;
+}
 
 /* exception types */
 PyObject* PyDockerfile_GoIOError;

